File: D1-Morning/D1-S2-PHP/W6-DataHiding/Account.php
<?php


class Account
{
    private $accountNumber;
    private $customerName;
    private $balance;

    public function __construct($accountNumber, $customerName)
    {
        $this->accountNumber = $accountNumber;
        $this->customerName = $customerName;
        $this->balance = 0;
    }

    public function getAccountNumber()
    {
        return $this->accountNumber;
    }

    public function setAccountNumber($accountNumber)
    {
        $this->accountNumber = $accountNumber;
    }

    public function getCustomerName()
    {
        return $this->customerName;
    }

    public function setCustomerName($customerName)
    {
        $this->customerName = $customerName;
    }

    public function getBalance()
    {
        return $this->balance;
    }


    public function deposit($amount){

        $this->balance = $this->balance+$amount;
        return $amount . " taka has been deposited.";
    }

    public function withdraw($amount){
        if($this->balance - $amount >= 0){
            $this->balance -= $amount;
            return $amount . " taka has been withdrawn.";
        }else{
            return "No sufficient balance";
        }
    }

}

File: D1-Morning/C#/D1-S2/W6-DataHiding/W6-DataHiding/Account.cs
namespace W6_DataHiding
{
    public class Account
    {
        public string AccountNumber { set; get; }
        public string CustomerName { set; get; }
        public double Balance { private set; get; }

        public Account(string accountNumber, string customerName):this()
        {
            AccountNumber = accountNumber;
            CustomerName = customerName;
        }

        public Account()
        {
            Balance = 0;
        }

        public string Deposit(double amount)
        {
            Balance += amount;
            return amount + " taka has been deposited."; 
        }

        public string Withdraw(double amount)
        {
            if (Balance - amount >= 0)
            {
                Balance -= amount;
                return amount + " taka has been withdrawn.";

            }
            else
            {
                return "No sufficient balance;";
            }
        }
    }
}
